pkg/polaris/polaris_test.go

```diff
@@ -397,6 +397,252 @@ func TestAzureSubscriptionAddAndRemove(t *testing.T) {
 	}
 }
 
+// TestAzureArchivalSubscriptionAddAndRemove verifies that the SDK can perform the
+// adding and removal of Azure subscription for archival feature
+// on a real Polaris instance.
+//
+// To run this test against a Polaris instance the following environment
+// variables needs to be set:
+//   * RUBRIK_POLARIS_SERVICEACCOUNT_FILE=<path-to-polaris-service-account-file>
+//   * TEST_INTEGRATION=1
+//   * TEST_AZURESUBSCRIPTION_FILE=<path-to-test-azure-subscription-file>
+//   * AZURE_AUTH_LOCATION=<path-to-azure-sdk-auth-file>
+//
+// The file referred to by TEST_AZURESUBSCRIPTION_FILE should contain a single
+// testAzureSubscription JSON object.
+func TestAzureArchivalSubscriptionAddAndRemove(t *testing.T) {
+	if !testsetup.BoolEnvSet("TEST_INTEGRATION") {
+		t.Skipf("skipping due to env TEST_INTEGRATION not set")
+	}
+
+	ctx := context.Background()
+
+	testSubscription, err := testsetup.AzureSubscription()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Add default Azure service principal to Polaris. Usually resolved using
+	// the environment variable AZURE_AUTH_LOCATION.
+	_, err = client.Azure().SetServicePrincipal(ctx, azure.Default(testSubscription.TenantDomain))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Add default Azure subscription to Polaris.
+	subscription := azure.Subscription(testSubscription.SubscriptionID, testSubscription.TenantDomain)
+	id, err := client.Azure().AddSubscription(
+		ctx,
+		subscription,
+		core.FeatureCloudNativeArchival,
+		azure.Regions("eastus2"),
+		azure.Name(testSubscription.SubscriptionName),
+		azure.ResourceGroup(testSubscription.Archival.ResourceGroupName, "eastus2", make(map[string]string)),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that the subscription was successfully added.
+	account, err := client.Azure().Subscription(
+		ctx,
+		azure.CloudAccountID(id),
+		core.FeatureCloudNativeArchival,
+	)
+	if err != nil {
+		t.Error(err)
+	}
+	if account.Name != testSubscription.SubscriptionName {
+		t.Errorf("invalid name: %v", account.Name)
+	}
+	if account.NativeID != testSubscription.SubscriptionID {
+		t.Errorf("invalid native id: %v", account.NativeID)
+	}
+	if account.TenantDomain != testSubscription.TenantDomain {
+		t.Errorf("invalid tenant domain: %v", account.TenantDomain)
+	}
+	if n := len(account.Features); n == 1 {
+		if name := account.Features[0].Name; name != "CLOUD_NATIVE_ARCHIVAL" {
+			t.Errorf("invalid feature name: %v", name)
+		}
+		if regions := account.Features[0].Regions; !reflect.DeepEqual(regions, []string{"eastus2"}) {
+			t.Errorf("invalid feature regions: %v", regions)
+		}
+		if account.Features[0].Status != "CONNECTED" {
+			t.Errorf("invalid feature status: %v", account.Features[0].Status)
+		}
+	} else {
+		t.Errorf("invalid number of features: %v", n)
+	}
+
+	// Update and verify regions for Azure account.
+	err = client.Azure().UpdateSubscription(ctx, azure.ID(subscription), core.FeatureCloudNativeArchival,
+		azure.Regions("westus2"))
```

Comment:
No, it doesn't require that.
We input resource group per (feature, subscription) and not per (feature, subscription, region) pair, so it should work in this case without any error.

It is different for encryption case, as it is a child feature, and parent feature is just archival(FeatureCloudNativeArchival), both of which are linked to same cloud account.
If we add more region to that subscription, it will be added to archival.
But if we create add encryption feature, that can only be added for the resource group in which managed identity is present(it should be one of region present for archival feature).

Reply:
:+1: